core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1700,6 +1700,39 @@ abstract class RDD[T: ClassTag](
     }
   }
 
+  /**
+   * :: Experimental ::
+   * Marks an RDD's shuffles and it's ancestors non-persisted ancestors as no longer needed.
+   * This cleans up shuffle files aggressively to allow nodes to be terminated.
+   * If you are uncertain of what you are doing please do not use this feature.
+   * Additional techniques for mitigating orphaned shuffle files:
+   *   * Tuning the driver GC to be more aggressive so the regular context cleaner is triggered
+   *   * Setting an appropriate TTL for shuffle files to be auto cleaned
+   */
+  @Experimental
+  @DeveloperApi
+  @Since("3.1.0")
+  def cleanShuffleDependencies(blocking: Boolean = false): Unit = {
+    sc.cleaner.foreach { cleaner =>
+      /**
+       * Clean the shuffles & all of its parents.
+       */
+      def cleanEagerly(dep: Dependency[_]): Unit = {
+        if (dep.isInstanceOf[ShuffleDependency[_, _, _]]) {
+          val shuffleId = dep.asInstanceOf[ShuffleDependency[_, _, _]].shuffleId
+          cleaner.doCleanupShuffle(shuffleId, blocking)
```

Comment:
Seems it's not thread safe to do so?

Reply:
I think it's fine, multiple concurrent cleanups would just log an error.

Reply:
It will also cause driver to send remove message to executors, though no fatal error there.

Reply:
Yes. Talking with @dbtsai he wanted to add a lock on the blocks inside of `doCleanupShuffle`, but given that the only price is duplicated messages to the executors I'm not sure its worth the overhead of keeping track of that many locks.

Reply:
How about this:

In `doCleanupShuffle`, we could call `if (MapOutputTrackerMaster.containsShuffle) {...}` before the cleaning.

Reply:
Yeah ok, get is fairly cheap on a `ConcurrentHashmap` I think that balances the tradeoffs well.